android/datastore.md

```diff
@@ -66,238 +65,245 @@ type Post @model {
+When the code generator builds model classes for your schema, it also generates a version hash based on the current schema. When you make changes to your schema and re-run the modelgen task, the code generator will update the hash if it detects any changes that will impact the underlying on-device storage structure. For example, a new hash will be generated if data types are added/deleted, or if any field has a change to its required/optional status. Whenever DataStore starts on a device, it will check for a change in the version hash. If the version has changed, all instances of these models will be wiped from the device, and if you are syncing with the cloud, a full sync will take place. Local migrations on device are not supported. If you are syncing with the cloud, the existing structure and instances of your data in the backend will not be modified as part of this process.
 
 ## Setup
 
-In your application's `onCreate` or similar lifecycle location initialize Amplify:
+In your app's `onCreate` (or similar lifecycle callback), initialize Amplify:
 
 ```java
-AWSDataStorePlugin dataStorePlugin = AWSDataStorePlugin.getInstance(AmplifyModelProvider.getInstance());
+AWSDataStorePlugin dataStorePlugin = AWSDataStorePlugin.forModels(AmplifyModelProvider.getInstance());
 Amplify.addPlugin(dataStorePlugin);
-Amplify.addPlugin(new AWSApiPlugin());
-Amplify.configure(context);
+Amplify.addPlugin(new AWSApiPlugin()); // If using remote model synchronization
+Amplify.configure(getApplicationContext());
+```
+
+If you do not have any configuration file at `app/src/main/res/raw/amplifyconfiguration.json`, create a placeholder for now. For more information about this file, see the configuration section of this guide.
+
+```
+{
+  "userAgent": "aws-amplify-cli/2.0",
+  "version": "1.0"
+}
 ```
 
 # Save Data
 
-To write any data to the DataStore you can pass an instance of a Model to `DataStore.save()` and it will be persisted in offline storage. At this point you can use it as an item in a normal data store such as querying, updating or deleting. If you choose to later connect to the cloud the item will be synchronized using GraphQL mutations and any other systems connected to the same backend can run queries or mutations on these items as well as observe them with GraphQL subscriptions.
+To write any data to the DataStore, you can pass an instance of a Model to `DataStore.save()` and it will be persisted in offline storage. The offline storage supports standard storage operations like querying, updating and deleting. If you choose to connect to the cloud at a later time, the local items will be synchronized using GraphQL mutations. Any other systems connected to the same backend can then run queries or mutations on your DataStore items, or observe changes to them via GraphQL subscriptions.
```

Comment:
Why delineate between offline storage? The point of DataStore is so that customers don't need to think about online vs offline, it's just operating on data.

Reply:
@undefobj Do you mean in this sentence?
> The offline storage supports standard storage operations like [...]

Agree; here, the distinction may not be helping. I'll make this into just `The DataStore supports standard storage operations like [...]`.